packages/table/src/Row.tsx

```diff
@@ -0,0 +1,373 @@
+import React, { useState, useEffect } from 'react';
+import { Transition } from 'react-transition-group';
+import Checkbox from '@leafygreen-ui/checkbox';
+import IconButton from '@leafygreen-ui/icon-button';
+import Icon from '@leafygreen-ui/icon';
+import { isComponentType } from '@leafygreen-ui/lib';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import { useTableContext, Types, DataType } from './TableContext';
+import Cell, { tdInnerDiv } from './Cell';
+
+const rowStyle = css`
+  border-top: 1px solid ${uiColors.gray.light2};
+  color: ${uiColors.gray.dark2};
+
+  & > td > ${tdInnerDiv.selector} {
+    height: 40px;
+    overflow: hidden;
+    transition: all 150ms ease-in-out;
+  }
+`;
+
+const altColor = css`
+  &:nth-of-type(even) {
+    background-color: ${uiColors.gray.light3};
+  }
+`;
+
+const iconButtonMargin = css`
+  margin-right: 4px;
+`;
+
+const disabledStyle = css`
+  background-color: ${uiColors.gray.light2};
+  color: ${uiColors.gray.base};
+  cursor: not-allowed;
+  border-top: 1px solid ${uiColors.gray.light1};
+  border-bottom: 1px solid ${uiColors.gray.light1};
+`;
+
+const displayFlex = css`
+  display: flex;
+  align-items: center;
+`;
+
+const truncation = css`
+  max-width: 100px;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+  overflow: hidden;
+`;
+
+const transitionStyles = {
+  default: css`
+    transition: border 150ms ease-in-out;
+    border-top-color: transparent;
+
+    & > td {
+      padding-top: 0px;
+      padding-bottom: 0px;
+    }
+
+    & > td > ${tdInnerDiv.selector} {
+      max-height: 0;
+    }
+  `,
+
+  entered: css`
+    border-top-color: ${uiColors.gray.light2};
+
+    & > td > ${tdInnerDiv.selector} {
+      max-height: 40px;
+    }
+  `,
+};
+
+const styleMap = {
+  left: [DataType.String, DataType.Weight, DataType.ZipCode, DataType.Date],
+  right: [DataType.Number, DataType.Quantity],
+} as const;
+
+function styleColumn(index: string, dataType: DataType) {
+  let justify = '';
```

Comment:
would this produce invalid css? e.g. `justify-content: ;`

Reply:
Good call, providing a fallback.